### DISCUSSION TITLE
Disable Linter

### DISCUSSION THREAD

**Discussion by joseneto** [BetterThanTomorrow/calva#1208] [General]
There is any option where I can disable linter?
This is really annoying most of my files has some unsolved symbols

**Comment by bpringe** [comment]
Hi. Linting is done via clojure-lsp. You can disable linting using clojure-lsp settings, see here: https://clojure-lsp.github.io/clojure-lsp/settings/#diagnostics-linter.

@ericdallo, does a user need to set both clj-kondo and clojure-lsp linters to `:off` in order to disable all linting? Something like this?

```edn
{:linters {:clj-kondo {:level :off}
           :unused-public-var {:level :off}}}
```

Do you think it would help to provide an example in the clojure-lsp docs specifically about disabling all linters?

@joseneto Instead of disabling linting, if those unresolved symbols are macros, you may want to configure clj-kondo to lint them as a built-in macro. See more information here: https://calva.io/linting. Then you can continue to benefit from linting.

Also, if you think something is wrong with the linting, please let us know.

**Reply by ericdallo** [reply]
yes @bpringe looks a good idea, I can improve that.

> @joseneto Instead of disabling linting, if those unresolved symbols are macros, you may want to configure clj-kondo to lint them as a built-in macro. See more information here: https://calva.io/linting. Then you can continue to benefit from listing.

Yes, I'd suggest following this path as well as clj-kondo linters are really good/helpful and customizable

**Reply by ericdallo** [reply]
[Done](https://clojure-lsp.github.io/clojure-lsp/settings/#disable-linter) @bpringe

**Reply by bpringe** [reply]
Awesome. Thanks.
